v3/src/components/graph/hooks/use-plot.ts

```diff
@@ -296,4 +276,13 @@ export const usePlotResponders = (props: IPlotResponderProps) => {
       {name: "respondToPointVisualChange", equals: comparer.structural}, graphModel
     )
   }, [callRefreshPointPositions, graphModel])
+
+  // respond to attribute color change
+  useEffect(function respondToColorChange() {
+    return mstReaction(
+      () => [metadata?.getLowColor(legendAttrID), metadata?.getHighColor(legendAttrID)],
```

Comment:
```suggestion
      () => metadata?.getColorRange(legendAttrID),
```
if you add that API.